eclair-core/src/main/scala/fr/acinq/eclair/Features.scala

```diff
@@ -87,7 +87,7 @@ case class Features(activated: Map[Feature, FeatureSupport], unknown: Set[Unknow
   def nodeAnnouncementFeatures(): Features = Features(activated.collect { case (f: NodeFeature, s) => (f: Feature, s) }, unknown)
 
   // NB: we don't include unknown features in invoices, which means plugins cannot inject invoice features.
-  def invoiceFeatures(): Map[Feature with InvoiceFeature, FeatureSupport] = activated.collect { case (f: InvoiceFeature, s) => (f, s) }
+  def invoiceFeatures(): Features = Features(activated.collect { case (f: InvoiceFeature, s) => (f: Feature, s) })
```

Comment:
Can you explain why you changed this? We now lose some type information: with the previous version we could ensure only `InvoiceFeature`s would be provided (and give that to constructors that only accept invoice features). It's true that when I changed that, I didn't change the constructor of `PaymentRequest` because I felt it didn't belong to the PR, I meant to refactor it later but now may be a good time :)